File: backend/src/main/kotlin/com/example/backend2/global/messaging/publisher/RedisMessagePublisher.java
package com.example.backend2.global.messaging.publisher;
import com.fasterxml.jackson.databind.ObjectMapper;
import lombok.RequiredArgsConstructor;
import org.example.bidflow.global.messaging.dto.MessagePayload;
import org.springframework.data.redis.core.RedisTemplate;
import org.springframework.stereotype.Component;

/**
 * Redis Pub/Sub 기반 구현체
 */

@Component
@RequiredArgsConstructor
public class RedisMessagePublisher implements MessagePublisher<MessagePayload> {

    private final RedisTemplate<String, Object> redisTemplate;

    @Override
    public void publish(String topic, MessagePayload payload) {
        String json = convertToJson(payload);
        redisTemplate.convertAndSend(topic, payload);
    }

    // 직렬화
    private String convertToJson(MessagePayload messagePayload) {
//        ObjectMapper objectMapper = new ObjectMapper();
//        return objectMapper.convertValue(messagePayload, String.class);
        return messagePayload.toString();
    }
}

File: backend/src/main/kotlin/com/example/backend2/global/messaging/publisher/RedisMessagePublisher.kt
package com.example.backend2.global.messaging.publisher

import com.example.backend2.global.messaging.dto.MessagePayload
import org.springframework.stereotype.Component

/**
 * Redis Pub/Sub 기반 구현체
 */
@Component
class RedisMessagePublisher(
    val redisTemplate: RedisTemplate<String, Any>,
) : MessagePublisher<MessagePayload?> {
    override fun publish(
        topic: String,
        message: MessagePayload,
    ) {
        val json = convertToJson(message)
        redisTemplate.convertAndSend(topic, json)
    }

    // 직렬화
    private fun convertToJson(messagePayload: MessagePayload): String {
//        ObjectMapper objectMapper = new ObjectMapper();
//        return objectMapper.convertValue(messagePayload, String.class);
        return messagePayload.toString()
    }
}
